plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs

```diff
@@ -593,180 +590,210 @@ evalBuiltinApp fun term runtime = case runtime of
     _ -> pure $ VBuiltin fun term runtime
 {-# INLINE evalBuiltinApp #-}
 
+data CekState uni fun =
+    -- the next state is computing
+    Computing WordArray (Context uni fun) (Closure uni fun)
```

Comment:
If we CPS it then these types will go away anyway. You probably also want to `{-# UNPACK #-}` at least the `Closure`.

Reply:
We don't care about performance that much so we may not go with CPS? 

Yes I should `{-# UNPACK #-}` thanks.

Reply:
I'm confused: I thought the idea was to refactor the _production_ CEK machine so that we can use it in both ways. In that case we _absolutely_ do care about the performance of the part used by both, no?

Reply:
maybe I'm confused......I'll push the ADR PR soon and we can discuss there.